Fix tools being consumed on processing

ehughsbaird opened the pull request:
#### Summary
Bugfixes "Fix EOC actor consuming tools on processing"

#### Purpose of change
Fixes https://github.com/CleverRaven/Cataclysm-DDA/issues/81719

#### Describe the solution
Allow the individual actors to specify whether or not they want to consume. Update all actors.

#### Testing
Spawn a chainsaw, activate it. It does not disappear.  Apply contact lenses, they are used.

#### Additional context
Why is there not a make noise iuse actor? These do not need a whole EOC for that.

ehughsbaird committed:
EOC iuse actor can choose to consume charges

Add a new member to the EOC use actor that determines whether or not it
will try to consume charges. Populate into relevant JSON fields.

ehughsbaird commented:
Yes, consume will consume on activation, assuming the activation code respects the return value of the activation function.

This is true for meds (I tested it on contacts and they worked), but the place_npc function always returns 1 (consume), so if that is not consuming the item it is a bug.

Maleclypse commented:
Definitely not consuming
<img width="1574" height="718" alt="NPCs forever" src="https://github.com/user-attachments/assets/e7a5c5b5-3231-4ee2-9faa-b7fb649c48c3" />
<img width="1589" height="747" alt="Make NPC" src="https://github.com/user-attachments/assets/a74e0c46-0876-488f-896f-548d46829ed4" />
<img width="1596" height="762" alt="frankenstein " src="https://github.com/user-attachments/assets/4254bd48-a015-492e-8249-edfc51af2a2b" />

ehughsbaird committed:
More correct return value handling for iuse

iuse functions return an optional<int>, where it is nullopt if the item
is not used, and the contained value is either how many charges were
used or if it wants to be destroyed - depending on the context.

Make the second behavior explicit with a flag, and apply it to the
molotov which depends on the destruction behavior.

RenechCDDA reviewed (approved):
The approach is reasonable, the flag is a good choice. The existing behavior is insidiously complicated due to charges.

I really appreciate this followup PR. I was expecting we might have one or two edge cases, but once again charges have ruined everything.

ehughsbaird committed:
Remove EPIC_MUSIC

Always returns std::nullopt since 9e6ea30c1f179b4d25e7d5e1afa75851c9fe0774

ehughsbaird committed:
Audit tick iuse return values

Charges are consumed through another method, so no tick iuse functions
should return that they use charges. Audit all the iuses used for tick
functions and ensure they do not return 1 (and still function
correctly).

ehughsbaird commented:
@Maleclypse I was mistaken about the behavior of these functions. You should add the `SINGLE_USE` flag to make items consumed on activation.
